doc/setup-development-environment.md

```diff
@@ -2,22 +2,120 @@
+Ubuntu 18.04 (Bionic Beaver). For other distributions you may consult the
+Dockerfile used by the integration tests. For example, 
+[Dockerfile.ubuntu](../ci/travis/Dockerfile.ubuntu), or
+[Dockerfile.fedora](../ci/travis/Dockerfile.fedora).
+
+First, install the basic development tools:
+
+```console
+sudo apt update
+sudo apt install -y build-essential cmake git gcc g++ cmake \
+        libc-ares-dev libc-ares2 libcurl4-openssl-dev libssl-dev make \
+        pkg-config tar wget zlib1g-dev
+```
+
+Then install `clang-6.0` and some additional Clang tools that we use to enforce
+code style rules:
+
+```console
+sudo apt install -y clang-6.0 clang-tidy clang-format clang-tools
+sudo update-alternatives --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-6.0 100
+sudo update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-6.0 100
+sudo update-alternatives --install /usr/bin/scan-build scan-build /usr/bin/scan-build-6.0 100
+```
+
+Install buildifier tool, which we use to format `BUILD` files:
+
+```console
+sudo wget -q -O /usr/bin/buildifier https://github.com/bazelbuild/buildtools/releases/download/0.17.2/buildifier
+sudo chmod 755 /usr/bin/buildifier
+```
+
+Install cmake_format to automatically format the CMake list files. We pin this
+tool to a specific version because the formatting changes when the "latest"
+version is updated, and we do not want the builds to break just
+because some third party changed something.
+
+```console
+sudo apt install -y python python-pip
+pip install --upgrade pip
+pip install numpy cmake_format==0.4.0
+```
+
+Install the Python modules used in the integration tests:
+
+```console
+pip install flask httpbin gevent gunicorn crc32c --user
+```
+
+Add the pip directory to your PATH:
+
+```console
+export PATH=$PATH:$HOME/.local/bin
+```
+
+You need to install the Google Cloud SDK. These instructions work for a GCE
+VM, but you may need to adapt them for your environment. Check the instructions
+on the
+[Google Cloud SDK website](https://cloud.google.com/sdk/) for alternatives.
+
+```console
+wget -q https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-233.0.0-linux-x86_64.tar.gz
+sha256sum google-cloud-sdk-233.0.0-linux-x86_64.tar.gz | \
+    grep -q '^a04ff6c4dcfc59889737810174b5d3c702f7a0a20e5ffcec3a5c3fccc59c3b7a '
+echo $?
+tar x -C $HOME -f google-cloud-sdk-233.0.0-linux-x86_64.tar.gz
+$HOME/google-cloud-sdk/bin/gcloud --quiet components install cbt bigtable
+```
+
+### Clone and compile `google-cloud-cpp`
+
+You may need to create a new key pair to connect to GitHub.  Search the web
+for how to do this.  Then you can clone the code:
+
+```console
+cd $HOME
+git clone git@github.com:<GITHUB-USERNAME_HERE>/google-cloud-cpp.git
+cd google-cloud-cpp
+git submodule update --init
+```
+
+And compile the code using:
+
+```console
+cmake -H. -B.build 
+cmake --build .build -- -j $(nproc)
+```
+
+Run the tests using:
+
+```console
+cd .build
```

Comment:
Should we use the "env -C" device here too?

Reply:
Yup, done.